packages/visitor-keys/src/visitor-keys.ts

```diff
@@ -212,7 +212,7 @@ const additionalKeys: AdditionalKeys = {
   TSIntersectionType: ['types'],
   TSIntrinsicKeyword: [],
   TSLiteralType: ['literal'],
-  TSMappedType: ['nameType', 'typeParameter', 'typeAnnotation'],
+  TSMappedType: ['constraint', 'key', 'nameType', 'typeAnnotation'],
```

Comment:
I think changing this order is what borked the scope analysis.

The ordering of keys should be the ordering of the nodes in the AST.

So it should be `['key', 'constraint', 'nameType', 'typeAnnotation']`

It looks like we used to have a bug in our code because the `nameType` is allowed to reference the `key` (eg `type U = {[T in string as T]: T}` is valid TS!)